astropy/modeling/functional_models.py

```diff
@@ -2062,3 +2078,87 @@ def evaluate(cls, x, y, amplitude, r_eff, n, x_0, y_0, ellip, theta):
         z = np.sqrt((x_maj / a) ** 2 + (x_min / b) ** 2)
 
         return amplitude * np.exp(-bn * (z ** (1 / n) - 1))
+
+
+class BlackBody1D(Fittable1DModel):
+    """
+    One dimensional blackbody model.
+
+    Parameters
+    ----------
+    temperature : `~astropy.units.Quantity`
+        Blackbody temperature
+    bolometric_flux : `~astropy.units.Quantity`
+        The bolometric flux of the blackbody (i.e. the integral over the
+        spectral axis)
+    """
+
+    # We parametrize this model with a temperature and a bolometric flux. The
+    # bolometric flux is the integral of the model over the spectral axis. This
+    # is more useful than simply having an amplitude parameter.
+    temperature = Parameter(default=5000 * u.K, min=0)
+    bolometric_flux = Parameter(default=1 * u.erg / u.cm ** 2 / u.s)
+
+    # We allow values without units to be passed when evaluating the model, and
+    # in this case the input x values are assumed to be frequencies in Hz.
+    input_units_allow_dimensionless = True
+
+    def evaluate(self, x, temperature, bolometric_flux):
+        """Evaluate the model.
+
+        Parameters
+        ----------
+        x : float or `~numpy.ndarray` or `~astropy.units.Quantity`
+            Frquency at which to compute the blackbody. If no units are given,
+            this defaults to Hz.
+
+        temperature : float or `~numpy.ndarray` or `~astropy.units.Quantity`
+            Temperature of the blackbody. If no units are given, this defaults
+            to Kelvin.
+
+        bolometric_flux : float or `~numpy.ndarray` or `~astropy.units.Quantity`
+            Desired integral for the blackbody.
+
+        Returns
+        -------
+        y : number or ndarray
+            Blackbody spectrum. The units are determined from the units of
+            ``bolometric_flux``.
+        """
+
+        from ..constants import sigma_sb
+        from ..analytic_functions.blackbody import blackbody_nu
+
+        # We need to make sure that we attach units to the temperature if it
+        # doens't have any units. We do this because even though blackbody_nu
+        # can take temperature values without units, the / temperature ** 4
+        # factor needs units to be defined.
+        temperature = u.Quantity(temperature, u.K)
+
+        # We normalize the returned blackbody so that the integral would be
+        # unity, and we then multiply by the bolometric flux. A normalized
+        # blackbody has f_nu = pi * B_nu / (sigma * T^4), which is what we
+        # calculate here. We convert to 1/Hz to make sure the units are
+        # simplified as much as possible, then we multiply by the bolometric
+        # flux to get the normalization right.
+        fnu = (np.pi * u.sr * blackbody_nu(x, temperature) /
```

Comment:
I'm afraid that I've been lost in the details, and can't see where u.sr is coming from?